equinix/resource_fabric_service_profile.go

```diff
@@ -1133,40 +1128,40 @@ func marketingInfoToFabric(schemaMarketingInfo []interface{}) v4.MarketingInfo {
 	return marketingInfos
 }
 
-func processStepToFabric(processStepl []interface{}) []v4.ProcessStep {
-	if processStepl == nil {
+func processStepToFabric(processSteps []interface{}) []v4.ProcessStep {
+	if processSteps == nil {
 		return []v4.ProcessStep{}
 	}
-	var processSteps []v4.ProcessStep
-	for _, processStep := range processStepl {
-		psTitle := processStep.(map[string]interface{})["title"].(interface{}).(string)
-		psSubTitle := processStep.(map[string]interface{})["sub_title"].(interface{}).(string)
-		psDescription := processStep.(map[string]interface{})["description"].(interface{}).(string)
-		processSteps = append(processSteps, v4.ProcessStep{
+	var processStepRes []v4.ProcessStep
+	for _, pcs := range processSteps {
+		psTitle := pcs.(map[string]interface{})["title"].(interface{}).(string)
+		psSubTitle := pcs.(map[string]interface{})["sub_title"].(interface{}).(string)
+		psDescription := pcs.(map[string]interface{})["description"].(interface{}).(string)
+		processStepRes = append(processStepRes, v4.ProcessStep{
 			Title:       psTitle,
 			SubTitle:    psSubTitle,
 			Description: psDescription,
 		})
 	}
-	return processSteps
+	return processStepRes
 }
 
 func portsToFabric(schemaPorts []interface{}) []v4.ServiceProfileAccessPointColo {
 	if schemaPorts == nil {
 		return []v4.ServiceProfileAccessPointColo{}
 	}
 	var ports []v4.ServiceProfileAccessPointColo
```

Comment:
You can either revert this method to how it was before, or I'm gonna ask for a lot of changes.

First, every `[]` variable should be made with `make` so that we can assign by index and elminate the extra processing needing by `append`.

All range variables should be `index, full_singular_name`.

All assignments should be done by index and not with `append`.